### ISSUE TITLE
Improvement on Units module

### ISSUE THREAD

**Issue by csguth** [eclufsc/ophidian#61]
I'm trying to create a proof of concept for the Ophidian global API, but I'm struggling on using the units module. I think the unit tests are not extensive enough and they also lack of documentation.

There are no coverage on math operations in general. For example:
- Squared units (e.g., area);
- Units conversion (e.g., capacitance*resistance to time).

I think the unit module is crucial for the 1.0 release.

**Comment by vinilivramento** [comment]
I agree that we need more unit tests. However some basic unit conversion is already there. There is also a unit test for time = capacitance * resistance.

**Comment by csguth** [comment]
@vinilivramento 

> However some basic unit conversion is already there.

Where? I can't see any unit test.

I have looked at `Units:40` and I saw something really bizarre:
```C++
//Area Units
using square_micrometer_t = units::unit<std::nano, units::category::length_unit>;
````
I'm pretty sure this line is semantically wrong. Since there is no test covering this line (at least I couldn't find it), I'm assuming this is completely wrong. Why `square_micrometer_t` is not defined in function of `square_meter_t`? 

Also, I'd like to see tests using [the unit literals](https://github.com/nholthaus/units/tree/15fc1069e60b98c9eb2ea28ebad3c74bdad1bd99#unit-literals).

**Comment by vinilivramento** [comment]
There are a few test units inside test/util/units_test.cpp
As I said previously, I agree that we need more unit tests. Only a few tests w.r.t. time, capacitance, resistance and Location were performed. We can definitely add more. 

Regarding the line you pointed out, yes there is something weird. I can't remember why I implemented in this way. I am gonna take a look and, if it is really wrong, I will fix it. 

I am not a big fan of unit literals, but we can add tests to ensure that this feature works well.
